packages/playground/CoreApp/App.cpp

```diff
@@ -0,0 +1,75 @@
+#include "pch.h"
+#include "CoreApp.h"
+
+// This app demonstrates the usage of the Microsoft.ReactNative CoreApp APIs
+// use the preprocessor defines WITH_MODULES (true/false), FROM_JSON (true/false),
+// or neither to exercise the RNStartCoreAppWithModules, RNStartCoreApp, and RNStartCoreAppFromConfigJson APIs.
+#define DEFINE_MODULES
+
+// #define WITH_MODULES
+#define FROM_JSON
+
+#ifdef DEFINE_MODULES
+#include <ModuleRegistration.h>
+#include <NativeModules.h>
+#include <winrt/Microsoft.ReactNative.h>
+
+REACT_MODULE(LogBox)
+struct LogBox {
+  REACT_INIT(Initialize)
+  void Initialize(const winrt::Microsoft::ReactNative::ReactContext &) noexcept {
+    MessageBox(nullptr, L"Native Modules work!", L"Success", MB_OK);
+  }
+};
+
+struct ThisAppPackageProvider
+    : winrt::implements<ThisAppPackageProvider, winrt::Microsoft::ReactNative::IReactPackageProvider> {
+  void CreatePackage(winrt::Microsoft::ReactNative::IReactPackageBuilder const &packageBuilder) noexcept {
+    for (auto const *reg = winrt::Microsoft::ReactNative::ModuleRegistration::Head(); reg != nullptr;
+         reg = reg->Next()) {
+      packageBuilder.AddModule(reg->ModuleName(), reg->MakeModuleProvider());
+    }
+  }
+};
+
+extern "C" __declspec(dllexport) void *__cdecl MySpecialPackageProvider() {
```

Comment:
>MySpecialPackageProvider [](http://example.com/codeflow?start=47&length=24)

It seems to be unused.

Reply:
this is exported by the app, and then app.config.json can be used to point to it (we'll discover it via LoadLibrary/GetProcAddress)

Reply:
I see. It makes sense.